docs/conventions/component.md

```diff
@@ -0,0 +1,461 @@
+
+### File name
+
+Twig file name should be the same as component's name.
+
+Example:
+```
+generic-component-navigation-menu.twig
+```
+
+### Generic component
+
+Generic component twig file defines all the markup for the component. It should contains all the logic required to use the component, as it is intended to be used by systems components.
+
+This twig file consists in different sections (see link above for more details):
+* Parameters/Blocks: list of exposed parameters and blocks
+* Internal properties: definition of custom variables
+* Internal logic: preprocess and other custom logic
+* Print: html markup
+
+Example:
+```twig
+{#
+  Parameters:
+  [...]
+
+  Blocks:
+  [...]
+#}
+
+{# Internal properties #}
+[...]
+
+{# Internal logic - Process properties #}
+[...]
+
+{# Print the result  #}
+[...]
+```
+
+### Systems component
+
+There are 2 possibilities for system based components: they use the exact same markup as generic component, or they alter it.
+
+In both cases, **twig file should always start with available parameters**.
+
+#### Same markup
+
+This is the easy case: system component just has to include generic twig file.
+
+Example:
+```twig
+{#
+  Parameters:
+  [...]
+
+  Blocks:
+  [...]
+#}
+
+{% include '@ecl/generic-component-navigation-menu' %}
+```
+
+#### Different markup
+
+As soon as there is a difference in markup (even a small one), specific component has to re-write the whole markup (instead of using generic one).
+This is the same structure as generic twig file.
+
+As we lose the advantage of generic components here, it is recommended, when possible, to keep the exact same markup for different systems.
+
+## Javascript
+
+Not all components require a dedicated javascript file. Most of them have simple enough behavior, that could be handled with html/css only. The following rules only concern component with more complex (js) behavior.
+
+Apart from [rules that should be applied to all js files](javascript), there are some specifications for components:
+
+### File name
+
+Javascript file name should be the same as component's name.
+
+Example:
+```
+generic-component-navigation-menu.js
+```
+
+### Generic component
+
+Generic component should export a function to handle specific behavior.
+This function should have an explicit name, based on component name (similar to css class).
+
+Example:
+```
+export const ecl-navigation-menu = ({ [...] } = {}) => {
+  [...]
+}
+
+export default ecl-navigation-menu;
+```
+
+### Systems component
+
+System components should use exported function from generic.
+
+Example:
+```
+export * from '@ecl/generic-component-navigation-menu';
+```
+
+## Config javascript file
+
+### File name
+
+Config file name should be the same as component's name.
+
+Example:
+```
+generic-component-navigation-menu.config.js
+```
+
+### Generic component
+
+There is no real config file for generic components, as they are not meant to be used directly.
```

Comment:
I think it's worth mentioning `*.config.js` are in transition to `context` folder as mentioned next, with the detail that these are the 2 ways of expressing components' model/data.